fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java

```diff
@@ -2221,4 +2254,112 @@ public void testPathsWithBrackets() {
         assertEquals(BAD_REQUEST.getStatusCode(), getStatus(postObjMethod("%5bfoo")));
         assertEquals(BAD_REQUEST.getStatusCode(), getStatus(putObjMethod("%5bfoo")));
     }
+
+    @Test
+    public void testBinaryLastModified() throws Exception {
+        final String objid = getRandomUniqueId();
+        final String objURI = serverAddress + objid;
+        final String binURI = objURI + "/binary1";
+
+        final long lastmod1;
+        try (final CloseableHttpResponse response = execute(putDSMethod(objid, "binary1", "some test content"))) {
+            assertEquals(CREATED.getStatusCode(), getStatus(response));
+            lastmod1 = headerFormat.parse(response.getFirstHeader("Last-Modified").getValue()).getTime();
+        }
+
+        sleep(1000); // wait a second to make sure last-modified value will be different
+
+        try (final CloseableGraphStore graph = getGraphStore(new HttpGet(binURI + "/fcr:metadata"))) {
+            verifyModifiedMatchesCreated(graph);
+        }
+
+        final HttpPatch patchBinary = new HttpPatch(binURI + "/fcr:metadata");
+        patchBinary.addHeader("Content-Type", "application/sparql-update");
+        patchBinary.setEntity(new StringEntity("INSERT { <" + binURI + "> " +
+                "<http://www.w3.org/TR/rdf-schema/label> \"this is a label\" } WHERE {}"));
+        final long lastmod2;
+        try (final CloseableHttpResponse response = execute(patchBinary)) {
+            assertEquals(NO_CONTENT.getStatusCode(), getStatus(response));
+            lastmod2 = headerFormat.parse(response.getFirstHeader("Last-Modified").getValue()).getTime();
+            assertTrue(lastmod2 > lastmod1);
+        }
+
+        sleep(1000); // wait a second to make sure last-modified value will be different
+
+        final long lastmod3;
+        try (final CloseableHttpResponse response = execute(putDSMethod(objid, "binary1", "new test content"))) {
+            assertEquals(NO_CONTENT.getStatusCode(), getStatus(response));
+            lastmod3 = headerFormat.parse(response.getFirstHeader("Last-Modified").getValue()).getTime();
+            assertTrue(lastmod3 > lastmod2);
+        }
+    }
+
+    @Test
+    public void testContainerLastModified() throws Exception {
+        final String objid = getRandomUniqueId();
+        final String objURI = serverAddress + objid;
+
+        // create an object
+        final long lastmod1;
+        try (final CloseableHttpResponse response = execute(putObjMethod(objid))) {
+            assertEquals(CREATED.getStatusCode(), getStatus(response));
+            lastmod1 = headerFormat.parse(response.getFirstHeader("Last-Modified").getValue()).getTime();
+        }
+
+        sleep(1000); // wait a second to make sure last-modified value will be different
```

Comment:
The FCREPO-2023 ticket is still relevant, but in this case, we'll still need to use `Thread.sleep`. I say that because this test is specifically about verifying that the reported last modified date occurs in a particular order corresponding to the order of operation. Because the reported last modified date is rounded to the nearest second, it is entirely possible that two operations in quick succession would produce last-modified dates that are equal, even if the event bus is used.